src/SFA.DAS.PR.Application/Permissions/Queries/GetHasPermissions/GetHasPermissionsQueryHandler.cs

```diff
@@ -0,0 +1,18 @@
+﻿using MediatR;
+using SFA.DAS.PR.Application.Mediatr.Responses;
+using SFA.DAS.PR.Domain.Interfaces;
+
+namespace SFA.DAS.PR.Application.Permissions.Queries.GetHasPermissions;
+
+public class GetHasPermissionsQueryHandler(IPermissionsReadRepository _permissionsReadRespository) : IRequestHandler<GetHasPermissionsQuery, ValidatedBooleanResult>
+{
+    public async Task<ValidatedBooleanResult> Handle(GetHasPermissionsQuery query, CancellationToken cancellationToken)
```

Comment:
Not sure why we need ValidatedBooleanResult, we could relax the type constraint on ValidatedResult so it can return bool? was this tried?

Reply:
@DevenKShah In order to return a bool with the current validated result structure we would need to encapsulate the bool in a class object as it is not a reference type. The ValidatedResponse object accepts TModel : class and therefore is expecting a reference type so cannot be used directly with a boolean.